Merge 4.45 code freeze to develop

oguzkocer opened the pull request.

oguzkocer committed:
Bump version number

oguzkocer committed:
Update draft release notes for 4.45.

oguzkocer committed:
Release Notes: add new section for next version (4.46)

oguzkocer committed:
Update strings for localization

oguzkocer committed:
Update release notes for 4.45

oguzkocer committed:
Update metadata strings

jkmassel commented on Simplenote/Resources/AppStoreStrings.pot:
```diff
@@ -15,12 +15,14 @@ msgstr ""
 #. translators: App name to be displayed in the Apple App Store. Limit to 30 characters including spaces and commas!
 msgctxt "app_store_title"
-msgid "Simplenote – Notes and Todos"
+msgid "Simplenote – Notes and Todos
```
This _may_ break the ability to import this file into GlotPress? Might be worth testing it with https://github.com/wordpress-mobile/docker-glotpress-linter?

mokagio replied:
I didn't test what the linter said. I just went ahead and fixed it in [`0a9932f` (#1435)](https://github.com/Automattic/simplenote-ios/pull/1435/commits/0a9932f73d5fb5b31d0aa4d347087dcd9acd10b9)

jkmassel commented on Simplenote/en.lproj/Localizable.strings:
```diff
@@ -1,857 +0,0 @@
-/* Number of found search results */
```
It seems surprising we would delete this entire file? I think something might've gone wrong here

mokagio replied:
Fixed by #1437.

oguzkocer commented:
@jkmassel Thanks for the review. Any chance you could help me with these issues? I have no idea what went wrong. The script didn't log any errors, so I thought everything was fine, but I agree that this seems off.

mokagio committed:
Update `find` usage in `localize.py` script to account for spaces in paths

e6a2a168a33c72be2050dffc9fcc65144655eac5
(https://github.com/Automattic/simplenote-ios/pull/1435/commits/e6a2a168a33c72be2050dffc9fcc65144655eac5)
deleted the entire `en.lproj/Localizable.strings` file because the
`find` command in `localize.py` returned a path with a space in it,
`./Simplenote/Supporting Files/main.m`. When parsing that space,
`genstring` failed:

```
genstrings: error: failed to read input file ./Simplenote/Supporting
```

I haven't dug deeper, but I'm guessing at some point there's a generic
"stage all changes in the folder" command for Git, which would have
resulted in the file being removed.

This commit makes use of the `-print0` option in `find` and of `| xargs
-0` to be robust against spaces in paths.

mokagio committed:
Update strings for localization

mokagio commented:
I doubt either of you would be working on this right now because of time zones, but just dropping a message to say that I found the localization issue and I'll be submitting a fix soon.

mokagio commented:
@jkmassel @oguzkocer @AliSoftware the file got deleted because we now have `./Simplenote/Supporting Files/main.m` which matches the query for files to feed as input to `genstrings` but, because of the space in the path, makes the command fail, with cascading results.

I opened two PRs with different takes on the fix:

- https://github.com/Automattic/simplenote-ios/pull/1436 updates `localize.py` to be more robust against spaces in paths
- https://github.com/Automattic/simplenote-ios/pull/1437 takes a different approach and removes the space from the path

~~I'm not sure which is best. Maybe both should be applied?~~ Actually, after further reviewing the diffs (see the "Compare" links in the description for both PRs) I was surprised to see that the output `.strings` file is different 🤔 The diff in #1437 seems more trust-worthy in that the other has a lot of deleted entries 🤔 

I'd be leaning towards using that for now and then spend more time investigating what's going on in `localize.py`. Maybe I was too hasty with the fix and I'd need to dive deeper in how the script uses `genstrings`.

mokagio commented:
Also notice that the UI tests are failing and that seems to be a consistent problem since a couple of days... You can see that they are failing on [a PR based of `develop`](https://github.com/Automattic/simplenote-ios/pull/1438), too, so I think we can ignore that failure here.

I'm following up on the issue, which I think is actually backend related, elsewhere.

pachlava committed:
Updates the test user(s) password to a more secure one

AliSoftware committed:
Fix find logic in localize.py

We need to use parentheses to be sure that the -print0 is applied to the whole expression

Otherwise only the RHS of the -o OR expression will have the -print0 applied, which means we will only print .swift files and not print .m files at all

AliSoftware commented:
@mokagio thanks for looking into it and for trying out the 2 solutions!

> Actually, after further reviewing the diffs (see the "Compare" links in the description for both PRs) I was surprised to see that the output .strings file is different 🤔 

I took a stab at it especially to try to understand why there was a difference in output in the 2 alternatives. I finally found the issue in your #1437 fix (missing parentheses to apply `-print0` to both `.m` and `.swift`), and after fixing that, #1437 ended up with the same diff for the generated `.lproj` as the diff in #1436.

Now that #1437 is fixed, I think we should go with that option (as opposed to #1436), as supporting spaces (instead of avoiding them) will be more resilient. Besides, as you said, `Supporting Files` (with a space) is indeed a common and standard name for this folder anyway, so there's a chance it could be accidentally changed back in the future by someone not realizing the consequence otherwise.

mokagio committed:
Account for spaces in input paths in `localize.py` script (#1437)

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

mokagio commented:
Thanks @AliSoftware 🙌 

As you can guess from my merge of #1437 into this branch, I do agree with your rationale (besides, this is yet another easy to change decision, so let's move ahead!)

mokagio commented on Simplenote/en.lproj/Localizable.strings:
```diff
@@ -163,7 +171,7 @@
 
 /* No comment provided by engineer. */
-"collaborators-description" = "Add an email address to share this note with someone. Then you can both make changes to it.";
+"collaborators-description" = "collaborators-description";
```
I'm concerned by this and I think we should fix it before merging:

<img width="426" alt="Screen Shot 2021-09-22 at 11 13 58 am" src="https://user-images.githubusercontent.com/1218433/134268601-d0dcb3ae-a560-47bb-a649-fc89b442bfd0.png">

I feel like it's an issue that existed already and that the improvement on how we run `genestrings` from #1437 revealed. 

Looking at the source, there is no comment in the `NSLocalizedString` macro:

https://github.com/Automattic/simplenote-ios/blob/bfa530c86e3cbd4fe89c01666de9f39351e83221/Simplenote/Classes/SPAddCollaboratorsViewController.m#L148

mokagio reviewed (commented):
I noticed the change in localization tooling resulted in a few strings changing in a way that is not correct.

We should address this before merging.

mokagio committed:
Update "Add a tag..." localized string definition in `SPTagEntryField.m`

mokagio committed:
Update the "collaborators-description" localized string definition

mokagio committed:
Updated "deleted note warning" localized string definition

mokagio committed:
Update the "add tag" localized accessibility hint definition

mokagio committed:
Update "remove tag" localized accessibility hint definition

mokagio committed:
Remove a couple of inappropriate newlines from `AppStoreStrings.pot`

This won't be necessary once we ship a new release toolkit version that
includes the fix from
https://github.com/wordpress-mobile/release-toolkit/pull/297

mokagio committed:
Update "Trash" noun localization definition in `NotesListFilter.swift`

mokagio committed:
Update "Trash" noun localization definition in `TagListViewController.swift`

mokagio committed:
Update strings for localization

mokagio commented:
> We should address [the generated strings not being human readable] before merging.

I opened https://github.com/Automattic/simplenote-ios/pull/1443 for this.

oguzkocer committed:
Merge pull request #1443 from Automattic/release/4.45-strings-fix

Update some localized string definitions to correct the English rendering
